### DISCUSSION TITLE
[@types/react-transition-group] nodeRef attribute has different type definition than React ref attribute

### DISCUSSION THREAD

**Discussion by goldmont** [DefinitelyTyped/DefinitelyTyped#59176] [Issues with a @types package]
Hi there,

I found an error in the types definition of [@types/react-transition-group](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-transition-group) package. The problem is that the attribute `nodeRef` ([Transition.d.ts #132](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-transition-group/Transition.d.ts#L132)) accepts a different data type than React `ref` attribute does ([index.d.ts #143](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L143)) Thus I'm forced to cast my ref instance to `any` type in order to make it acceptable by the `nodeRef` attribute.

Ref
```typescript
const ref = useRef<HTMLElement | null>(null);
```

React (works)
```
ref?: LegacyRef<T> | undefined;
```

React Transition Group (doesn't work)
```typescript
nodeRef?: React.Ref<RefElement> | undefined;
```

Error:
`Type MutableRefObject<HTMLElement | null> is not assignable to type React.Ref<Ref> | undefined`

Thanks.

[a comment by an automated account is not shown]

**Comment by bengry** [comment]
I think it's safe to change the type in the pakchae package to be `LegacyRef`. Feel free to submit a PR, just make sure to add tests to see that both the existing uses don't break.
